showcase.yaml

```diff
@@ -1,219 +1,214 @@
+          Action:
+          - iam:ListUsers
+          - iam:GetGroup
           Resource: '*'
         - Effect: Allow
           Action:
-          - 'iam:ListSSHPublicKeys'
-          - 'iam:GetSSHPublicKey'
-          Resource: !Sub 'arn:aws:iam::${AWS::AccountId}:user/*'
+          - iam:ListSSHPublicKeys
+          - iam:GetSSHPublicKey
+          Resource: !Sub arn:aws:iam::${AWS::AccountId}:user/*
         - Effect: Allow
-          Action: 'ec2:DescribeTags'
+          Action: ec2:DescribeTags
           Resource: '*'
       Roles:
         - !Ref Role
+
   Instance:
     Type: AWS::EC2::Instance
-    Metadata:
-      'AWS::CloudFormation::Init':
-        config:
-          packages: !If
-          - UseUbuntu
-          - apt:
-              git: []
-              awscli: []
-          - yum:
-              git: []
-          files:
-            '/opt/install.sh':
-              source: 'https://raw.githubusercontent.com/widdix/aws-ec2-ssh/master/install.sh'
-              mode: '000755'
-              owner: root
-              group: root
-          commands:
-            a_install:
-              command: !Sub './install.sh -a "${AssumeRole}"'
-              cwd: '/opt'
     Properties:
       ImageId: !FindInMap [!FindInMap [OSMap, !Ref OS, RegionMap], !Ref 'AWS::Region', AMI]
       IamInstanceProfile: !Ref InstanceProfile
-      InstanceType: 't2.micro'
+      InstanceType: t2.micro
       KeyName: !If [HasKeyName, !Ref KeyName, !Ref 'AWS::NoValue']
-      UserData: !If
-      - UseUbuntu
-      - 'Fn::Base64': !Sub |
-          #!/bin/bash -x
-          bash -ex << "TRY"
-            apt-get update
-            apt-get -y install python-setuptools
-            mkdir aws-cfn-bootstrap-latest
-            curl -s -m 60 https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz | tar xz -C aws-cfn-bootstrap-latest --strip-components 1
-            easy_install aws-cfn-bootstrap-latest
-            /usr/local/bin/cfn-init --verbose --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-          TRY
-          /usr/local/bin/cfn-signal --exit-code $? --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-      - 'Fn::Base64': !Sub |
-          #!/bin/bash -x
-          /opt/aws/bin/cfn-init --verbose --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
-          /opt/aws/bin/cfn-signal --exit-code $? --stack=${AWS::StackName} --region=${AWS::Region} --resource=Instance
+      UserData:
+        Fn::Base64: !Sub |
+          #cloud-config
+
+          packages:
+            # Install packages needed to install the cloudformation bootstrap from source
+            # (doing this to be more cross platform compatible)
+            - python-setuptools
+            - curl
+            - tar
+
+          runcmd:
+            # Install cfn tools
+            - easy_install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
```

Comment:
Same issue here as with the ```aws``` executable above. This installs another version as comes with the os, probably also in another directory.

Reply:
Similar comments to above - bootstrap is not installed by default on AmazonLinux (I believe), so the end result may be a different directory than yum would have used, however it's a consistent directory on Ubuntu and AmazonLinux which was my goal.